src/jobs/closeUnsubmittedRenewalsCheck.ts

```diff
@@ -0,0 +1,161 @@
+    const endTime = new Date();
+    const jobSuccessReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: endTime,
+      success: true,
+      details,
+    };
+    logger.info(`${JOB_NAME} - Report: ${JSON.stringify(jobSuccessReport)}`);
+    return jobSuccessReport;
+  } catch (err) {
+    logger.error(`${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`);
+    const jobEndTime = new Date();
+    const jobFailedReport: JobReport<BatchJobDetails> = {
+      jobName: JOB_NAME,
+      startedAt: jobStartTime,
+      finishedAt: jobEndTime,
+      success: false,
+      error: `${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`,
+    };
+    logger.error(`${JOB_NAME} - Report: ${JSON.stringify(jobFailedReport)}`);
+    return jobFailedReport;
+  }
+}
+
+// update app state, including transition to CLOSED + update event
+const closeApplication = async (
+  currentApp: Application,
+  identity: Identity,
+): Promise<Application> => {
+  const appObj = new ApplicationStateManager(currentApp);
+  const role = getDacoRole(identity);
+  logger.info(`${JOB_NAME} - Role ${role} is trying to CLOSE appId ${currentApp.appId}.`);
+  const result = appObj.updateApp({ state: 'CLOSED' }, false, {
+    id: identity.userId,
+    role,
+  });
+  logger.info(`${JOB_NAME} - Updating ${result.appId} in db to ${result.state}.`);
+  const updatedApp = await ApplicationModel.findOneAndUpdate({ appId: result.appId }, result, {
+    new: true,
+  }).exec();
+  if (updatedApp) {
+    return updatedApp;
+  } else {
+    throw new Error(`${JOB_NAME} - Find and update operation failed for ${currentApp.appId}.`);
+  }
+};
+
+export const getAppsClosingQuery = (currentDate: Date): FilterQuery<ApplicationDocument> => {
+  const referenceDate = moment(currentDate).utc().startOf('day');
+  const query: FilterQuery<ApplicationDocument> = {
+    state: {
+      $in: ['DRAFT', 'SIGN AND SUBMIT', 'REVISIONS REQUESTED'],
+    },
+    renewalPeriodEndDateUtc: { $lt: moment(referenceDate).startOf('day').toDate() },
+    isRenewal: true,
+  };
+  return query;
+};
+
+const getClosedAppsReportDetails = async (
+  user: Identity,
+  currentDate: Date,
+): Promise<BatchJobDetails> => {
+  const query = getAppsClosingQuery(currentDate);
+  const closableAppCount = await ApplicationModel.find(query).countDocuments();
+  // if no applications fit the criteria, return empty report details
+  if (closableAppCount === 0) {
+    logger.info(`${JOB_NAME} - No renewal applications need to be closed at this time.`);
+    logger.info(`${JOB_NAME} - Generating report.`);
+    return getEmptyReportDetails();
+  }
+  logger.info(`${JOB_NAME} - There are ${closableAppCount} apps that should be CLOSED.`);
+  const closableApps = await ApplicationModel.find(query).exec();
+
+  const doCloseApplication = async (app: ApplicationDocument): Promise<JobResultForApplication> => {
```

Comment:
Is there a reason this method is declared inside another function instead of at root of this module?

Reply:
It's purely lazy copying on my part - we did this pattern for the pause app job but looking at it it doesn't seem to need to be defined there...